build/files.js

```diff
@@ -231,6 +231,22 @@ function CorkedRequest(state) {
       /Buffer\.prototype\.copy\.call\(src, target, offset\);/
     , 'src.copy(target, offset);'
   ]
+  , maybeDefineProperty = [
+        headRegexp
+      , '$1\n' + `
+/* <replacement> */
+var maybeDefineProperty = Object.defineProperty;
+// IE < 9
+if (!maybeDefineProperty || global.document && global.document.documentMode < 9) {
```

Comment:
Looks right as far as I'm aware; `document.documentMode` is `11` on the last IE unless I fiddle with the compatibility mode, in which case it becomes `8` on compatibility mode for IE 8. Moreover, the fix did seem to work for IE7, which is even lower.

Running this bit of code (the `if` condition with `!maybeDefineProperty` replaced with `!Object.defineProperty`) in compatibility mode for 7, 8 and 9 got `true` for 7 and 8 and `false` for 9.

There's always brute-force try-catch-based feature sniffing to deal with the fact that IE 8 has the function but it doesn't work right; but that would probably be overkill considering this works fine as far as I can tell.